### PR TITLE
FEATURE: Provide basic documentation about Eel

### PR DESCRIPTION
**Pull request by DanielSiepmann**
* To explain what Eel is about.
* To ease starting with Eel.
* To provide a central place for Eel where further documentation is
  referenced.

### CODE REVIEW

**Comment by DanielSiepmann** [comment]
Rendered result can be viewed at http://tmp.daniel-siepmann.de/docs/flow/eel/TheDefinitiveGuide/PartIII/Eel.html

**Comment by kitsunet** [comment]
Looks good to me. For the Contexts a sentence about the `ProtectedContextAwareInterface`  could be added, but that can be done afterwards. Not a blocker for this PR.

**Comment by kitsunet** [comment]
Merging in 24h is no objections.

File: TYPO3.Flow/Documentation/TheDefinitiveGuide/PartIII/Eel.rst [hunk 1]
```diff
@@ -0,0 +1,124 @@
+.. _eel:
+
+===
+Eel
+===
+
+.. sectionauthor:: Daniel Siepmann <coding@daniel-siepmann.de>
+
+Eel stands for Embedded Expression Language and enabled developers to create a Domain Specific
```
Comment (dimaip) [inline TYPO3.Flow/Documentation/TheDefinitiveGuide/PartIII/Eel.rst]: s/enabled/enables/

File: TYPO3.Flow/Documentation/TheDefinitiveGuide/PartIII/Eel.rst [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+the context needed to evaluate the expression.
+
+An expression can be something like::
+
+    'foo.bar == "Test" || foo.baz == "Test" || reverse(foo).bar == "Test"'
+
+To enable this expression to be parsed, a context is needed to define ``foo.bar``, ``foo.baz`` and
+``reverse()``.
+
+Basically a context is nothing more then an array defining the parts as key value pars. The above
```
Comment (dimaip) [inline TYPO3.Flow/Documentation/TheDefinitiveGuide/PartIII/Eel.rst]: s/pars/pairs/

File: TYPO3.Flow/Documentation/TheDefinitiveGuide/PartIII/Eel.rst [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+                'baz' => 'Test2',
+            ],
+            'reverse' => function ($array) {
+                return array_reverse($array, true);
+            }
+        ]);
+        $result = (new CompilingEvaluator)->evaluate($expression, $context);
+
+In the above example ``$result`` will be a boolean. But the result depends on the expression and can
+be any type.
```
Comment (dimaip) [inline TYPO3.Flow/Documentation/TheDefinitiveGuide/PartIII/Eel.rst]: of any type

**Comment by DanielSiepmann** [comment]
Thx @dimaip I've update the commit according to you suggestions.

**Comment by albe** [comment]
Merging with previous votes. Very nice, was really missing all along!

**Pull request merged**